doc/syslog/syslog-design.md

```diff
@@ -0,0 +1,559 @@
+            "vrf": "default"
+        },
+        "3.3.3.3": {
+            "source": "1.1.1.1",
+            "port": "514",
+            "vrf": "mgmt"
+        },
+        "2222::2222": {
+            "source": "1111::1111",
+            "port": "514",
+            "vrf": "Vrf-Data"
+        }
+    }
+}
+```
+
+### 2.4.4 Configuration migration
+
+No special handling is required
+
+## 2.5 Flows
+
+### 2.5.1 SSIP add
+
+![SSIP add flow](images/ssip_add_flow.svg "Figure 1: SSIP add flow")
+
+###### Figure 1: SSIP add flow
+
+### 2.5.2 SSIP remove
+
+![SSIP remove flow](images/ssip_remove_flow.svg "Figure 2: SSIP remove flow")
+
+###### Figure 2: SSIP remove flow
+
+## 2.6 CLI
+
+### 2.6.1 Command structure
+
+**User interface**:
+```
+config
+|--- syslog
+     |--- add <server_ip> OPTIONS
+     |--- del <server_ip>
+
+show
+|--- syslog
+```
+
+**Options:**
+
+_config syslog add_
+1. `-s|--source` - source ip address
+2. `-p|--port` - server udp port
+3. `-r|--vrf` - vrf device
+
+### 2.6.2 Usage examples
+
+#### 2.6.2.1 Config command group
+
+**The following command adds/deletes syslog server:**
+```bash
+config syslog add '2.2.2.2' \
+--source '1.1.1.1' \
+--port '514' \
+--vrf 'default'
+config syslog del '2.2.2.2'
+```
+
+#### 2.6.2.2 Show command group
+
+**The following command shows syslog server configuration:**
+```bash
+root@sonic:/home/admin# show syslog
+SERVER      SOURCE      PORT    VRF
+----------  ----------  ------  --------
+2.2.2.2     1.1.1.1     514     default
+3.3.3.3     1.1.1.1     514     mgmt
+2222::2222  1111::1111  514     Vrf-Data
+```
+
+## 2.7 YANG model
+
+An existing YANG model `sonic-syslog.yang` will be extended in order to provide support for SSIP.
+
+**Skeleton code:**
+```yang
+module sonic-syslog {
+
+    yang-version 1.1;
+
+    namespace "http://github.com/Azure/sonic-system-syslog";
+    prefix syslog;
+
+    import ietf-inet-types {
+        prefix inet;
+    }
+
+    import sonic-vrf {
+        prefix vrf;
+    }
+
+    description "Syslog YANG Module for SONiC OS: remote syslog logging";
+
+    revision 2022-04-18 {
+        description "Initial revision.";
+    }
+
+    typedef vrf-device {
+        description "Represents syslog VRF device";
+        type enumeration {
+            enum default;
+            enum mgmt;
+        }
+    }
+
+    container sonic-syslog {
+
+        container SYSLOG_SERVER {
+
+            description "SYSLOG_SERVER part of config_db.json";
+
+            list SYSLOG_SERVER_LIST {
+
+                key "server_address";
+
+                leaf server_address {
+                    description "Syslog server IP address";
+                    type inet:ip-address;
+                }
+
+                leaf source {
+                    description "Syslog source IP address";
+                    type inet:ip-address;
+                }
+
+                leaf port {
+                    description "Syslog server UDP port";
+                    type uint16;
```

Comment:
Can we use inet:port-number? and default value 514?

Reply:
@venkatmahalingam a long time ago there was a discussion regarding a default values in a YANG models and Community decided not to use them as there might be different values for different vendors. Has something changed since that time?